feat(*): ssr-friendlify components (#2222)
Kong/kongponents: docs/guide/contributing.md

#### Prop naming

It's essential to choose context-aware and descriptive names. This practice ensures that the purpose and usage of each prop are clear to other developers. Additionally, avoid using prefixed props, as they can lead to redundancy and confusion. For example, instead of using `hasError`, simply use `error`, and replace `isSelected` with `selected`. This approach not only simplifies the prop names but also enhances the readability and maintainability of our codebase.

#### Attributes

Sometimes you will need to generate a random string to be used as value for various attributes (e.g. accessibility-related attributes like `id`, `for`, `aria-labelledby`, etc.). 

To generate a unique id so that it is safe for SSR, you **must** use the `useUniqueId` composable in your component:

> ```ts

Reviewer: ```suggestion
```html
```

TODO: wip

#### Attributes

> Sometimes you will need to generate a random string to be used as value for various attributes (e.g. accessibility-related attributes like `id`, `for`, `aria-labelledby`, etc.). To generate a random id you can use a `useUniqueId` composable:

Reviewer: ```suggestion
Sometimes you will need to generate a random string to be used as value for various attributes (e.g. accessibility-related attributes like `id`, `for`, `aria-labelledby`, etc.). 

To generate a random id so that it is safe for SSR, you **must** use the `useUniqueId` composable in your component:
```

```ts
<script setup lang="ts">
import useUniqueId from '@/composables/useUniqueId'

const id = useUniqueId()
</script>
```

> Then you need to use [`v-bind-once` directive](https://github.com/danielroe/vue-bind-once) to bind any attributes that could potentially produce hydration issues in SSR mode:

Reviewer: ```suggestion
Then you must to use the [`v-bind-once` directive](https://github.com/danielroe/vue-bind-once) to bind the unique id to element attributes (this prevents hydration issues in SSR apps):
```

TODO: wip

> #### Attributes

Reviewer: docs: can you remove the "TODO: wip" above?

Author: Wrote a paragraph about prop naming instead

Sometimes you will need to generate a random string to be used as value for various attributes (e.g. accessibility-related attributes like `id`, `for`, `aria-labelledby`, etc.). 

> To generate a random id so that it is safe for SSR, you **must** use the `useUniqueId` composable in your component:

Reviewer: ```suggestion
To generate a unique id so that it is safe for SSR, you **must** use the `useUniqueId` composable in your component:
```

```ts
<script setup lang="ts">
import useUniqueId from '@/composables/useUniqueId'

const id = useUniqueId()
</script>
```

> Then you must to use the [`v-bind-once` directive](https://github.com/danielroe/vue-bind-once) to bind the unique id to element attributes (this prevents hydration issues in SSR apps):

Reviewer: ```suggestion
Then you must use the [`v-bind-once` directive](https://github.com/danielroe/vue-bind-once) to bind the unique id to element attributes (this prevents hydration issues in SSR apps):
```
